test/Chainweb/Test/Pact/PactMultiChainTest.hs

```diff
@@ -233,6 +234,52 @@ chainweb213Test = do
         , "(create-table tbl)"
         ]
 
+pactLocalDepthTest :: PactTestM ()
+pactLocalDepthTest = do
+  runToHeight 53
+  runBlockTest
+    [ PactTxTest (buildCoinXfer "(coin.transfer 'sender00 'sender01 1.0)") $
+        assertTxGas "Coin transfer pre-fork" 1583
+    ]
+  runBlockTest
+    [ PactTxTest (buildCoinXfer "(coin.transfer 'sender00 'sender01 1.0)") $
+        assertTxGas "Coin post-fork" 1583
+    ]
+
+  runLocalWithDepth (Just $ BlockHeight 0) cid getSender00Balance >>= \r ->
+    checkLocalResult r $ assertTxSuccess "Should get the current balance" (pDecimal 99999997.6834)
+
+  -- checking that `Just $ BlockHeight 0` has the same behaviour as `Nothing`
+  runLocalWithDepth Nothing cid getSender00Balance >>= \r ->
+    checkLocalResult r $ assertTxSuccess "Should get the current balance as well" (pDecimal 99999997.6834)
+
+  runLocalWithDepth (Just $ BlockHeight 1) cid getSender00Balance >>= \r ->
+    checkLocalResult r $ assertTxSuccess "Should get the balance one block before" (pDecimal 99999998.8417)
+
+  runLocalWithDepth (Just $ BlockHeight 2) cid getSender00Balance >>= \r ->
+    checkLocalResult r $ assertTxSuccess "Should get the balance two blocks before" (pDecimal 100000000)
+
+  -- the negative depth turns into 18446744073709551611 and we expect the `LocalRewindLimitExceeded` exception
+  -- since `BlockHeight` is a wrapper around `Word64`
+  handle (\(LocalRewindLimitExceeded _ _) -> return ()) $ do
+    runLocalWithDepth (Just $ BlockHeight (-5)) cid getSender00Balance >>= \_ ->
+      liftIO $ assertFailure "block succeeded"
+
+  -- depth that reaches the genesis
+  runLocalWithDepth (Just $ BlockHeight 99) cid getSender00Balance >>= \r ->
+    checkLocalResult r $ assertTxSuccess "Should get the balance at the genesis block" (pDecimal 100000000)
```

Comment:
Two tests — with a negative depth, which tests the limit, and a depth reaching the genesis block.